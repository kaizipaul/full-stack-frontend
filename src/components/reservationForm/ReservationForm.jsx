import React, { useState } from 'react';
import Select from 'react-select';
import Modal from '../modal/Modal';
import './reservationForm.css';

const cities = [
  { value: 'New York', label: 'New York' },
  { value: 'Los Angeles', label: 'Los Angeles' },
  { value: 'Oregon', label: 'Oregon' },
  { value: 'Bradenton Beach', label: 'Bradenton Beach' },
  { value: 'Charlottetown', label: 'Charlottetown' },
  { value: 'Bangkog', label: 'Bangkog' },
  { value: 'Beijing', label: 'Beijing' },
];

const ReservationForm = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [selectedCity, setSelectedCity] = useState('');

  const handleModalOpen = () => {
    setIsModalOpen(true);
  };

  const handleModalClose = () => {
    setIsModalOpen(false);
  };

  return (
    <div className="container">
      <div className="header">
        <div className="nav">
          <div className="nav-icon">
            <i className="fas fa-bars" />
          </div>
          <div className="search-icon">
            <i className="fas fa-search" />
          </div>
        </div>
        <div className="title">
          <h4>BOOK A TEST DRIVE WITH WHEEL-WIZARD</h4>
          <hr className="divider" />
          <p style={{ color: '#fff' }}>
            Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ducimus, blanditiis. lorem10
          </p>
        </div>
        <div className="buttons" style={{ display: 'flex', alignItems: 'center', marginTop: '2rem' }}>
          <Select
            id="select_city"
            className="select"
            options={cities}
            value={selectedCity}
            onChange={setSelectedCity}
            placeholder="Select a City..."
            required
            styles={{
              control: (provided) => ({
                ...provided,
                borderRadius: '25px',
                height: '2.7rem',
                background: '#a2d31a',
                border: '1px solid white',
                width: '10.2rem',
                color: 'white',
              }),
              placeholder: (provided) => ({
                ...provided,
                color: 'white',
              }),
              dropdownIndicator: (provided) => ({
                ...provided,
                color: 'white',
              }),
              menu: (provided) => ({
                ...provided,
                zIndex: 100,
                width: '200px',
              }),
            }}
          />

          <button
            id="book_now"
            type="button"
            style={{ marginLeft: '1rem' }}
            onClick={handleModalOpen}
          >
            Book Now
          </button>
        </div>

      </div>
      {isModalOpen && (
        <div className="modal">
          <div className="modal-content">
            <button className="cls-m-btn" type="button" onClick={handleModalClose}>X</button>
            <br />
            <Modal selectedCity={selectedCity.value} setIsModalOpen={setIsModalOpen} />
          </div>
        </div>
      )}
    </div>
  );
};

export default ReservationForm;
