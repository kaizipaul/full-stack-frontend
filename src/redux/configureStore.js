import { configureStore, combineReducers } from '@reduxjs/toolkit';
// import { carsReducer } from './cars/carsView';
import reservationReducer from './reservation/reserve';
import createCarReducer from './newCar/car';
import deleteCarReducer from './deleteCar/deleteCar';
import { authReducer } from './auth/auth';

const token = localStorage.getItem('token');
const initialState = {
  auth: {
    token: token || null,
    isAuthenticated: !!token,
  },
};

const rootReducer = combineReducers({
  // cars: carsReducer,
  newcar: createCarReducer,
  auth: authReducer,
  delete: deleteCarReducer,
  reservations: reservationReducer,
});

export default configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});
