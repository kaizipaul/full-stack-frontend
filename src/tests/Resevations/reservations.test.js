import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../../redux/configureStore';
import ReservationForm from '../../components/reservationForm/ReservationForm';

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: 'mock data' })),
}));

describe('Reservations', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={configureStore}>
        <BrowserRouter>
          <ReservationForm />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
