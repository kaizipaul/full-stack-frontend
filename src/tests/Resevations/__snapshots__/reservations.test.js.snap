// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header"
        >
          <div
            class="nav"
          >
            <div
              class="nav-icon"
            >
              <i
                class="fas fa-bars"
              />
            </div>
            <div
              class="search-icon"
            >
              <i
                class="fas fa-search"
              />
            </div>
          </div>
          <div
            class="title"
          >
            <h4>
              BOOK A TEST DRIVE WITH WHEEL-WIZARD
            </h4>
            <hr
              class="divider"
            />
            <p
              style="color: rgb(255, 255, 255);"
            >
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ducimus, blanditiis. lorem10
            </p>
          </div>
          <div
            class="buttons"
            style="display: flex; align-items: center; margin-top: 2rem;"
          >
            <div
              class="select css-b62m3t-container"
              id="select_city"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class=" css-1k4xn47-control"
              >
                <div
                  class=" css-1fdsijx-ValueContainer"
                >
                  <div
                    class=" css-1550uzu-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select a City...
                  </div>
                  <div
                    class=" css-qbdosj-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-2-placeholder"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-required="true"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class=""
                      id="react-select-2-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1u9des2-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-1stb430-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <button
              id="book_now"
              style="margin-left: 1rem;"
              type="button"
            >
              Book Now
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div
          class="nav"
        >
          <div
            class="nav-icon"
          >
            <i
              class="fas fa-bars"
            />
          </div>
          <div
            class="search-icon"
          >
            <i
              class="fas fa-search"
            />
          </div>
        </div>
        <div
          class="title"
        >
          <h4>
            BOOK A TEST DRIVE WITH WHEEL-WIZARD
          </h4>
          <hr
            class="divider"
          />
          <p
            style="color: rgb(255, 255, 255);"
          >
            Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ducimus, blanditiis. lorem10
          </p>
        </div>
        <div
          class="buttons"
          style="display: flex; align-items: center; margin-top: 2rem;"
        >
          <div
            class="select css-b62m3t-container"
            id="select_city"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1k4xn47-control"
            >
              <div
                class=" css-1fdsijx-ValueContainer"
              >
                <div
                  class=" css-1550uzu-placeholder"
                  id="react-select-2-placeholder"
                >
                  Select a City...
                </div>
                <div
                  class=" css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-required="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1stb430-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <button
            id="book_now"
            style="margin-left: 1rem;"
            type="button"
          >
            Book Now
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
